monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala

```diff
@@ -0,0 +1,121 @@
+    if (Platform.isJVM)
+      Test.Parameters.default.withMaxSize(256)
+    else
+      Test.Parameters.default.withMaxSize(32)
+  }
+
+  def naiveImp[F[_], A](iter: Iterant[F, A], takeEveryNth: Int)(implicit F: Sync[F]): Iterant[F, A] =
+    iter
+      .zipWithIndex
+      .flatMap {
+        case (a, idx) =>
+          if ((idx + 1) % takeEveryNth == 0)
+            Iterant[F].pure(a)
+          else
+            Iterant[F].empty
+      }
+
+  test("naiveImp smoke test") { implicit s =>
+    val input = List(1, 2, 3, 4, 5, 6)
+    val iter = Iterant[Coeval].fromList(input)
+    assertEquals(naiveImp(iter, 1).toListL.value, input)
+    assertEquals(naiveImp(iter, 2).toListL.value, List(2, 4, 6))
+    assertEquals(naiveImp(iter, 3).toListL.value, List(3, 6))
+    assertEquals(naiveImp(iter, input.length + 1).toListL.value, List.empty[Int])
+  }
+
+  test("Iterant[Task].takeEveryNth equivalence with naiveImp") { implicit s =>
+    check3 { (list: List[Int], idx: Int, nr: Int) =>
+      val stream = arbitraryListToIterant[Task, Int](list, math.abs(idx) + 1, allowErrors = false)
+      val length = list.length
+      // scale down from (Int.MinValue to Int.MaxValue) to (1 to (length + 1)) range
+      val n = math.round(
+        (length * (nr.toDouble - Int.MinValue.toDouble)) / (Int.MaxValue.toDouble - Int.MinValue.toDouble) + 1
+      ).toInt
+      val actual = stream.takeEveryNth(n).toListL.runAsync
+      val expected = naiveImp(stream, n).toListL.runAsync
+      s.tick()
+      actual.value <-> expected.value
+    }
+  }
+
+  test("Iterant.takeEveryNth protects against broken batches") { implicit s =>
+    check1 { (iter: Iterant[Task, Int]) =>
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Task].nextBatchS[Int](new ThrowExceptionBatch(dummy), Task.now(Iterant[Task].empty), Task.unit)
+      val stream = iter.onErrorIgnore ++ suffix
+      val received = stream.takeEveryNth(1)
+      received <-> Iterant[Task].haltS[Int](Some(dummy))
+    }
+  }
+
+  test("Iterant.takeEveryNth protects against broken cursors") { implicit s =>
+    check1 { (iter: Iterant[Task, Int]) =>
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Task].nextCursorS[Int](new ThrowExceptionCursor(dummy), Task.now(Iterant[Task].empty), Task.unit)
+      val stream = iter.onErrorIgnore ++ suffix
+      val received = stream.takeEveryNth(1)
+      received <-> Iterant[Task].haltS[Int](Some(dummy))
+    }
+  }
+
+  test("Iterant.takeEveryNth triggers early stop on exception") { _ =>
+    check1 { (iter: Iterant[Coeval, Int]) =>
+      val cancelable = BooleanCancelable()
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Coeval].nextCursorS[Int](new ThrowExceptionCursor(dummy), Coeval.now(Iterant[Coeval].empty), Coeval.unit)
+      val stream = (iter.onErrorIgnore ++ suffix).doOnEarlyStop(Coeval.eval(cancelable.cancel()))
+      intercept[DummyException] { stream.takeEveryNth(1).toListL.value }
```

Comment:
Here you need to be precise about how the error gets thrown too. So the test should be:

```scala
assertEquals(stream.takeEveryNth(1).toListL.runTry, Failure(dummy))
```